keras_cv/metrics/object_detection/box_coco_metrics_test.py

```diff
@@ -139,41 +139,23 @@ def test_name_parameter(self):
             )
 
     def test_coco_metric_suite_ragged_prediction(self):
-        bounding_boxes = {
-            "boxes": tf.ragged.constant(
-                [
-                    [[10, 10, 20, 20], [100, 100, 150, 150]],  # small, medium
-                    [
-                        [200, 200, 400, 400],  # large
-                    ],
-                ],
-                dtype=tf.float32,
-            ),
-            "classes": tf.ragged.constant(
-                [
-                    [0, 1],
-                    [2],
-                ],
-                dtype=tf.float32,
-            ),
-            "confidence": tf.ragged.constant(
-                [
-                    [0.7, 0.8],
-                    [0.9],
-                ],
-                dtype=tf.float32,
-            ),
-        }
-        dense_bounding_boxes = bounding_box.to_dense(bounding_boxes)
-        ragged_bounding_boxes = bounding_box.to_ragged(dense_bounding_boxes)
         suite = BoxCOCOMetrics(bounding_box_format="xyxy", evaluate_freq=1)
-        y_true = dense_bounding_boxes
-        y_pred = ragged_bounding_boxes
+        y_true, y_pred, _ = load_samples(SAMPLE_FILE)
+        y_true = bounding_box.to_dense(y_true)
+        y_pred = bounding_box.to_ragged(y_pred)
```

Comment:
For a ragged test, I think it's best if we use inputs that are actually ragged in the sense that they don't happen to have all the same shape long the ragged dimension. We could do that by either updating the sample boxes or just using some manual new data (either adding it to the sample data or creating pure manual data).

Because we already assert that the numerics are correct in another test, it's not critical to use the goldens in this test case.

Reply:
I have updated the unit test to use manually created data with different shape along the ragged dimension.

And it actually caught a bug. Please see below.